Benqi-Agents/Large-Pool-Withdrawal/src/agent.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Finding,
+  FindingType,
+  FindingSeverity,
+  TransactionEvent,
+  HandleTransaction,
+  getEthersProvider,
+} from "forta-agent";
+
+import { ethers } from "ethers";
+
+import { COMPTROLLER_IFACE, QITOKEN_IFACE } from "./abi";
+
+import { createFinding } from "./finding";
+
+const COMPTROLLER_ADDR = "0x486Af39519B4Dc9a7fCcd318217352830E8AD9b4";
+
+const THRESHOLD_PERCENTAGE = 0.25;
+
+// Array to track the QiToken pools
+let QITOKENS: Set<string> = new Set<string>();
+
+export const getTotalSupply = async (qiToken: string, blockNumber: any) => {
+  // Create an ethers contract instance to query on-chain data
+  const qiTokenContract = new ethers.Contract(
+    qiToken,
+    QITOKEN_IFACE.format(ethers.utils.FormatTypes.full),
+    getEthersProvider()
+  );
+  try {
+    return await qiTokenContract.totalSupply({ blockTag: blockNumber });
+  } catch (error) {
+    // Not sure what to do when there's an error
```

Comment:
Thoughts on what to do when there is an error in the Web3 call?
Should we assume it's within the threshold or outside the threshold?

Reply:
Not sure about this, it would make more sense to consider it below the threshold, to avoid giving false alerts. 
But I believe it's important to point out that the agent is running into an issue, otherwise it will never give alerts. 
Maybe returning an `error Finding` in that case (and for all our agents) is a good idea, the finding will include a description of the problem encountered (ex. `Agent unable to fetch the totalSupply from qiToken contract`).

Reply:
why do we need to 'fetch' total supply in the first place? isn't it a specific number that will never change?

Reply:
The naming structure is quite confusing but these qiTokens are different to QI tokens.
See [qiToken](https://docs.benqi.fi/resources/contracts/benqi-liquidity-markets) addresses vs [QI Token](https://docs.benqi.fi/resources/contracts/benqi-liquidity-markets) address.

Reply:
As for considering failures, after a discussion with Lazaro I will be removing the `try/catch` from the code as I have only written it to protect against network issues rather than any potential wrong calls.